### PR TITLE
MHR Home Certification

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#12533

*Description of changes:*
* Implements Home Certification Components on Step 1 and Step 5
* Includes shared DatePicker Component
* Updated input filed composables and implementations
* Started migration of DateMixin to a Date Composable

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/composables/useInputRules.ts [hunk 1]
```diff
@@ -1,45 +1,40 @@
...
-      v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
-      v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+      v => (v ? /^\d+$/g.test(v) : true) || 'Invalid number'
     ]
   }
 
-  const maxLengthRules = (maxLength: number): Array<Function> => {
-    return [v => v.length <= maxLength || `Maximum ${maxLength} characters`]
+  /** Create a custom rules array use predefined rules. **/
+  const customRules = (...rules: any) => {
```
Comment (dimak1) [inline ppr-ui/src/composables/useInputRules.ts:29]: Nice 👍🏻

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeCertification.vue [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      minDate: computed(() => {
+        // Determined by YEAR value in Manufacturers, Make, Model Section
+        const utcDate = createUtcDate(getMhrRegistrationHomeDescription.value?.baseInformation.year, 0, 1)
+        return dateToYyyyMmDd(utcDate)
+      })
+    })
+
+    /** Apply local models to store when they change. **/
+    watch(() => localState.certificationOption, () => {
+      setMhrHomeCertification({ key: 'certificationOption', value: localState.certificationOption })
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/YourHome/HomeCertification.vue:177]: A clean way to reuse the same setter. 👍🏻

**Review by dimak1 — APPROVED** [review]
Looks good! Approved (awaiting next commit).

**Review by doug-lovett — APPROVED** [review]
LGTM.

File: ppr-ui/src/composables/useInputRules.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/composables/useInputRules.ts:29]: Thanks!

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeCertification.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/YourHome/HomeCertification.vue:177]: Less is more 😁

**Comment by cameron-eyds** [comment]
> Looks good! Approved (awaiting next commit).

Cool, thanks! Tests added 👍

**Pull request merged**